lib/systemd/journal/navigable.rb

```diff
@@ -137,11 +137,21 @@ def with_auto_reopen
         if auto_reopen
           @sd_call_count += 1
           if @sd_call_count >= auto_reopen
-            cursor = self.cursor
+            begin
+              cursor = self.cursor
+            rescue
+              # Cancel the reopen process if cursor method causes 'Cannot assign requested address' error
+              @sd_call_count = 0
+              return
+            end
+
+            matches = @reopen_filterable_matches.dup
 
             close
             initialize(@reopen_options)
 
+            filter(matches)
```

Comment:
It's not obvious to me that this will apply the exact same filters, depending on how they are applied.  For example if I call
```
add_filter(...)
add_conjunction
add_filter(...)
```

and the journal is re-opened, will they be applied in the same way? I'm not sure but I don't think so.

Reply:
Thanks. Indeed, seems that the filter using add_disjunction cannot be reapplied.
sounds on_open/on_reopen hooks are reasonable.

Reply:
Fixed with https://github.com/ledbettj/systemd-journal/pull/109/commits/2a90791304e05f7656049ae035f0735dc5f5ea41

If you find this difficult to maintain, I will consider using hooks.